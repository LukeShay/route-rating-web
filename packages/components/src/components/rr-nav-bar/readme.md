# rr-nav-bar

Navigation bar to be used in routerating. This component is used as the parent
to `rr-nav-link`. This component has a `ul` component containing the children
you pass in.

<!-- Auto Generated Below -->


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
