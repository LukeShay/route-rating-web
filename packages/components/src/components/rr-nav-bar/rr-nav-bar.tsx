import { Component, Element, Host, State, h } from '@stencil/core'

@Component({ tag: 'rr-nav-bar' })
export class RrNavBar {
  /**
   * Stores the children of this select
   */
  @State() children: any[] = []

  /**
   * Stores a reference to this element.
   */
  @Element() host: HTMLRrNavBarElement

  componentWillLoad(): void {
    this.children = Array.from(this.host.children)
  }

  render() {
    return (
      <Host>
        <ul>
          <slot />
          {this.children.map((child) => child.outerHTML)}
        </ul>
      </Host>
    )
  }
}
