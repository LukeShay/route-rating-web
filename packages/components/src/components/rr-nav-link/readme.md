# rr-nav-link

Navigation link to be used with routerating. This component is used as a child
of `rr-nav-bar`. This component contains a `li` component with a
`stencil-route-link` inside.

<!-- Auto Generated Below -->


## Properties

| Property           | Attribute | Description                                 | Type     | Default     |
| ------------------ | --------- | ------------------------------------------- | -------- | ----------- |
| `url` _(required)_ | `url`     | The relative URL where this should link to. | `string` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
