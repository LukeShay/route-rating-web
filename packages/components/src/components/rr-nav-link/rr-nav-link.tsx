import { Component, Element, Host, Prop, State, h } from '@stencil/core'

@Component({ tag: 'rr-nav-link' })
export class RrNavLink {
  /**
   * The relative URL where this should link to.
   */
  @Prop() url!: string

  /**
   * Stores the children of this select
   */
  @State() children: any[] = []

  /**
   * Stores a reference to this element.
   */
  @Element() host: HTMLRrNavLinkElement

  componentWillLoad(): void {
    this.children = Array.from(this.host.children)
  }

  render() {
    return (
      <Host>
        <li>
          <stencil-route-link url={this.url} activeClass="active" exact={true}>
            <slot />
            {this.children.map((child) => child.outerHTML)}
          </stencil-route-link>
        </li>
      </Host>
    )
  }
}
