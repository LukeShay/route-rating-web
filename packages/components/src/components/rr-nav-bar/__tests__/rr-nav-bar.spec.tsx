import { RrNavBar } from '../rr-nav-bar'
import { RrNavLink } from '../../rr-nav-link/rr-nav-link'
import { newSpecPage } from '@stencil/core/testing'

describe('rr-nav-bar', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [RrNavBar, RrNavLink],
      html: `<rr-nav-bar><rr-nav-link url="asdf">URL</rr-nav-link></rr-nav-bar>`,
    })

    expect(page.root).toMatchSnapshot()
  })
})
