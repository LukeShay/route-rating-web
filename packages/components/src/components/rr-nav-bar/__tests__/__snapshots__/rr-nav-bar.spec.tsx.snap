// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rr-nav-bar should match snapshot 1`] = `
<rr-nav-bar>
  <!---->
  <ul>
    <rr-nav-link url="asdf">
      <!---->
      <li>
        <stencil-route-link activeclass="active" exact="" url="asdf">
          URL
        </stencil-route-link>
      </li>
    </rr-nav-link>
    &lt;rr-nav-link url="asdf"&gt;&lt;!----&gt;URL&lt;/rr-nav-link&gt;
  </ul>
</rr-nav-bar>
`;
